packages/ui/src/components/InputFields/SmartInputField/use-smart-input-field.ts

```diff
@@ -11,96 +11,98 @@ export type IdentifierInputType =
   | SignInIdentifier.Phone
   | SignInIdentifier.Username;
 
-export type EnabledIdentifierTypes = IdentifierInputType[];
+export type IdentifierInputValue = {
+  type: IdentifierInputType | undefined;
+  value: string;
+};
 
 const digitsRegex = /^\d*$/;
 
 type Props = {
   defaultValue?: string;
-  onChange?: (value: string) => void;
-  enabledTypes: EnabledIdentifierTypes;
-  currentType: IdentifierInputType;
-  onTypeChange?: (type: IdentifierInputType) => void;
+  defaultType?: IdentifierInputType;
+  enabledTypes: IdentifierInputType[];
 };
 
-const useSmartInputField = ({
-  onChange,
-  currentType,
-  enabledTypes,
-  onTypeChange,
-  defaultValue,
-}: Props) => {
-  const { countryCode: defaultCountryCode, inputValue: defaultInputValue } = parseIdentifierValue(
-    currentType,
-    defaultValue
+const useSmartInputField = ({ defaultType, defaultValue, enabledTypes }: Props) => {
+  const enabledTypeSet = useMemo(() => new Set(enabledTypes), [enabledTypes]);
+
+  if (defaultType) {
+    assert(
+      enabledTypeSet.has(defaultType),
+      new Error(
+        `Invalid input type. Current inputType ${defaultType} is detected but missing in enabledTypes`
+      )
+    );
+  }
+
+  // Parse default type from enabled types if default type is not provided and only one type is enabled
+  const _defaultType = useMemo(
```

Comment:
nit: change the "defaultType" in component's property to "_defaultType", and use "defaultType" here.

Reply:
updated